src/main/webapp/pipe.js

```diff
@@ -216,20 +223,18 @@ function pipelineUtils() {
                 Q('#pipeline-message-' + pipelineid).html('');
             }
 
-            var index = 0;
             var source;
             var target;
             lastResponse = data;
             equalheight('.pipeline-row .stage');
 
-            Q.each(data.pipelines, function (i, component) {
-                Q.each(component.pipelines, function (j, pipeline) {
-                    index = j;
-                    Q.each(pipeline.stages, function (k, stage) {
+            Q.each(data.pipelines, function (dataPipelineIndex, component) {
```

Comment:
@tommysdk 
Confirm is this OK renaming of one-symbol and two-symbol to appropriate name.